### PR TITLE
Make metrics reusable by other remappers

### PR DESCRIPTION
**Pull request by lahsivjar**
`AddMetrics` function would be required by other mappers, migrate it to an internal package to allow this.

### CODE REVIEW

File: remappers/internal/metric.go [hunk 1]
```diff
@@ -23,46 +23,49 @@ import (
...
 
-		dp.SetTimestamp(metric.timestamp)
-		if metric.startTimestamp != 0 {
-			dp.SetStartTimestamp(metric.startTimestamp)
+		dp.SetTimestamp(metric.Timestamp)
+		if metric.StartTimestamp != 0 {
+			dp.SetStartTimestamp(metric.StartTimestamp)
 		}
 
 		dp.Attributes().PutStr("event.provider", "hostmetrics")
```
Comment (ishleenk17) [inline remappers/internal/metric.go:71]: Lets add the change to event.module in this PR itself. 
```suggestion
		dp.Attributes().PutStr("event.module", "elastic/opentelemetry-lib")
```

**Comment by ishleenk17** [comment]
Changes look good. Left one comment, after addressing that we can merge it.
@lahsivjar : I hope we have tested it as well after reorganizing the structure.

File: remappers/internal/metric.go (hunk 1 above)
Comment (lahsivjar) [reply remappers/internal/metric.go:71]: I prefer to create a separate PR for that.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**